legacy/src/main/java/org/opensearch/sql/legacy/utils/QueryDataAnonymizer.java

```diff
@@ -39,7 +39,7 @@ public static String anonymizeData(String query) {
                     .replaceAll("[\\n][\\t]+", " ");
         } catch (Exception e) {
             LOG.warn("Caught an exception when anonymizing sensitive data");
-            resultQuery = query;
+            resultQuery = "";
```

Comment:
The problem with returning an empty string is that it's not solving the root of the problem.  The anonymizer is repressing ALL exceptions from the anonymizer, and returning a valid state.  
We should, instead, be returning a failure state in the response or throwing some exception.  The function that's calling the anonymizer is completely unaware that an error occurred.